src/command_modules/azure-cli-advisor/azure/cli/command_modules/advisor/commands.py

```diff
@@ -3,53 +3,25 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
-from azure.cli.core.commands import cli_command
 from azure.cli.command_modules.advisor._client_factory import \
     (advisor_mgmt_client_factory,
      recommendations_mgmt_client_factory,
      suppressions_mgmt_client_factory,
      configurations_mgmt_client_factory)
-
-custom_path = 'azure.cli.command_modules.advisor.custom#'
-
-cli_command(
-    __name__,
-    'advisor recommendation generate',
-    custom_path + 'cli_advisor_generate_recommendations',
-    recommendations_mgmt_client_factory
-)
-
-cli_command(
-    __name__,
-    'advisor recommendation list',
-    custom_path + 'cli_advisor_list_recommendations',
-    recommendations_mgmt_client_factory
-)
-
-cli_command(
-    __name__,
-    'advisor recommendation disable',
-    custom_path + 'cli_advisor_disable_recommendations',
-    suppressions_mgmt_client_factory
-)
-
-cli_command(
-    __name__,
-    'advisor recommendation enable',
-    custom_path + 'cli_advisor_enable_recommendations',
-    advisor_mgmt_client_factory  # using advisor client here because this spans recommendations and suppressions
-)
-
-cli_command(
-    __name__,
-    'advisor configuration get',
-    custom_path + 'cli_advisor_get_configurations',
-    configurations_mgmt_client_factory
-)
-
-cli_command(
-    __name__,
-    'advisor configuration set',
-    custom_path + 'cli_advisor_set_configurations',
-    configurations_mgmt_client_factory
-)
+from ._validators import validate_include_or_exclude, validate_ids_or_resource_group
+
+
+def load_command_table(self, _):
+    with self.command_group('advisor recommendation') as g:
+        g.custom_command('generate', 'cli_advisor_generate_recommendations',
+                         client_factory=recommendations_mgmt_client_factory)
+        g.custom_command('list', 'cli_advisor_list_recommendations', validator=validate_ids_or_resource_group,
```

Comment:
Why is this a command validator?  It seems like it would simply be a parameter level validator.

Reply:
How do we differentiate between parameter and command level validators? 
This validator is dependent on two different parameters, hence why I put it on the command level.

Reply:
Just because a validator depends on two different parameters doesn't make it a command-level validator.  Command level validators are used when you need multiple validators and the sequence matters. These sound to me like parameter validators. Also, by convention, command level validators have typically had names like "process_foo_namespace" but that's just a convention.

Reply:
What would happen when you have two different parameter level validators that both act on multiple arguments?